### ISSUE TITLE
Map variables are fully overwritten instead of keeping unmentioned keys

### ISSUE THREAD

**Issue by mfiano** [SuperCuber/dotter#29]
Is the following expected behavior?

global.toml
```toml
# snip
[graphics.files]
"xinitrc" = "~/.xinitrc"
[graphics.variables.display]
xrandr = false
primary_device = ""
primary_mode = "1920x1080"
primary_position = "0x0"
secondary_device = ""
secondary_mode = "1920x1080"
secondary_position = "0x0"
# snip
```

local.toml
```toml
packages = ["graphics"]

[variables.display]
xrandr = true
primary_device = "HDMI1"
secondary_device = "eDP-1"
```

xinitrc
```sh
{{#if display.xrandr}}
xrandr \
  --output {{display.secondary_device}} \
  --off \
  --pos {{display.secondary_position}} \
  --output {{display.primary_device}} \
  --primary \
  --mode 1920x1080 \
  --pos {{display.primary_position}}
{{/if}}
```

```sh
$ dotter
[ERROR] Failed to update template template "xinitrc" -> "/home/mfiano/.xinitrc"
        Caused by:
            perform template deployment
            render template
            Error rendering "Unnamed template" line 8, col 9: Variable "display.secondary_position" not found in strict mode.
            Error rendering "Unnamed template" line 8, col 9: Variable "display.secondary_position" not found in strict mode.
```

As a couple of unrelated side notes, as to not bombard your issues:

1) I don't really like that the templating engine does not remove the condition lines in the output. In the above example, if it compiles the file, there would be an additional newline where `{{#if display.xrandr}}` and `{{/if}}` are specified in the input. I think this just contributes to long configuration files with a lot of conditionalized template substitution.

2) I am not a fan of handlebars, because in two ocassions, I was unable to record a file due to an ambiguity in the syntax of that file and handlebars syntax. `{{` is a common delimiter. Have a look at chezmoi for example, which uses `{{-`.

**Comment by SuperCuber** [comment]
Thanks for the issue, this is not expected behavior.

For 1, take a look at how I'm doing it in [my zshrc](https://github.com/SuperCuber/dotfiles/blob/68ae7e70f1b81d0f35ffaf04b31f16c423714114/zshrc) for example - the `~` means "remove all whitespace following directly this". I put it at the end of each part of the conditional syntax as the rule of thumb.\
Also see [this issue](https://github.com/sunng87/handlebars-rust/issues/258) from handlebars project.

For 2, you can use [complex targets](https://github.com/SuperCuber/dotter/wiki/Setup-and-Configuration#complex-target) to force a whole file to be a symlink and [raw blocks](https://docs.rs/handlebars/3.5.1/handlebars/#built-in-helpers) to ignore parts of templates. Feel free to open a feature request issue if you have other ideas for this.

---

A question for you as a user - is there anything you find missing/suboptimal in the documentation? You can give feedback in #10

**Comment by SuperCuber** [comment]
Fixed in 9f3b55f8fcc4428f311bdd4223e29d546e124b42

**Comment by SuperCuber** [comment]
A release with this fix has been published
